release/src/main/scripts/build_release_candidate.sh

```diff
@@ -112,31 +113,12 @@ if [[ $confirmation = "y" ]]; then
   fi
   mkdir -p ${LOCAL_CLONE_DIR}
   cd ${LOCAL_CLONE_DIR}
-  git clone ${GIT_REPO_URL}
+  git clone --depth 1 --branch "${RC_TAG}" ${GIT_REPO_URL} "${BEAM_ROOT_DIR}"
   cd ${BEAM_ROOT_DIR}
-  git checkout ${RELEASE_BRANCH}
-  RELEASE_COMMIT=$(git rev-parse --verify ${RELEASE_BRANCH})
 
   echo "-------------Building Java Artifacts with Gradle-------------"
   git config credential.helper store
 
-  if git rev-parse "v${RELEASE}-RC${RC_NUM}" >/dev/null 2>&1; then
-    echo "Tag v${RELEASE}-RC${RC_NUM} already exists."
-    echo "Delete the tag and create a new tag commit (y) or skip this step (n)? [y/N]"
-    read confirmation
-    if [[ $confirmation = "y" ]]; then
-      # Delete tag with the git push <from>:<to> format, as shown here:
-      # https://git-scm.com/docs/git-push#Documentation/git-push.txt-codegitpushoriginexperimentalcode
-      git push origin :refs/tags/v${RELEASE}-RC${RC_NUM}
```

Comment:
Retroactive CC @emilymye that I removed the editing of the repo from the RC building. RC commit selection and RC building are now totally separate. When an RC commit is chosen it is still on SNAPSHOT/dev versions. There is a single off-branch commit tagged and pushed that updates these versions.